### DISCUSSION TITLE
The memory block 'Elsa.Expressions.Models.MemoryBlockReference' does not exist error.

### DISCUSSION THREAD

**Discussion by galgodage** [elsa-workflows/elsa-core#4904] [Design & tech]
I execute workflow from json . Here is json.
{
"id": "48f3ccdbc8e7a9c7",
"definitionId": "787d88475a179b36",
"name": "Child Workflow",
"createdAt": "2024-02-04T19:22:54.0083211+00:00",
"version": 1,
"toolVersion": "3.0.0.0",
"variables": [],
"inputs": [
{
"uiHint": "single-line",
"storageDriverType": "Elsa.Workflows.Services.WorkflowStorageDriver, Elsa.Workflows.Core",
"type": "Object",
"name": "test",
"displayName": "test",
"category": "Primitives",
"isArray": false
}
],
"outputs": [],
"outcomes": [],
"customProperties": {
"Elsa:WorkflowContextProviderTypes": []
},
"isReadonly": false,
"isLatest": true,
"isPublished": false,
"options": {
"autoUpdateConsumingWorkflows": false
},
"root": {
"type": "Elsa.Flowchart",
"version": 1,
"id": "477edcca49777c8f",
"nodeId": "Workflow1:477edcca49777c8f",
"metadata": {},
"customProperties": {
"source": "FlowchartJsonConverter.cs:45",
"notFoundConnections": [],
"canStartWorkflow": false,
"runAsynchronously": false
},
"activities": [

{
"text": {
"typeName": "String",
"expression": {
"type": "JavaScript",
"value": "getInput('test')"
},

  "memoryReference": {
    "id": "WriteLine1:test"
  }
},
"id": "bc9e6ebb7c7b3f74",
"nodeId": "Workflow1:477edcca49777c8f:bc9e6ebb7c7b3f74",
"name": "WriteLine1",
"type": "Elsa.WriteLine",
"version": 1,
"customProperties": {
  "canStartWorkflow": false,
  "runAsynchronously": false
},
"metadata": {
  "designer": {
    "position": {
      "x": -780,
      "y": 679.8888893127441
    },
    "size": {
      "width": 139.296875,
      "height": 50
    }
  }
}
}
],
"connections": [
{
"source": {
"activity": "595ac7b8405a0957",
"port": "Done"
},
"target": {
"activity": "bc9e6ebb7c7b3f74",
"port": "In"
}
}
]
}
}

I think error comming from this block

"expression": {
"type": "JavaScript",
"value": "getInput('test')"
},

  "memoryReference": {
    "id": "WriteLine1:test"
  }
this retun The memory block 'Elsa.Expressions.Models.MemoryBlockReference' does not exist error.
Please my deadline is passing. give help to solve this

**Comment by glime-ai** [comment]
Based on the JSON you've provided and the error message, it seems like there might be an issue with the way the `getInput` function is being called or how the memory reference is set up in the `WriteLine` activity.

In Elsa, to access input variables from JavaScript, you should use the `input` object followed by the name of the variable. For example, if you have an input variable named `test`, you would access it in a JavaScript expression like this: `input.test`.

Here's how you might correct the JavaScript expression:

```json
"expression": {
  "type": "JavaScript",
  "value": "input.test"
}
```

Please note that the `memoryReference` part in your JSON seems to be incorrect or unnecessary for accessing the input variable. Memory references are typically used to reference specific memory locations in Elsa, and it's not clear from the provided context how `WriteLine1:test` is intended to be used.

If you're still encountering issues, please ensure that the input named `test` is being passed correctly to the workflow when it's being executed. If the input is not set or passed incorrectly, it could result in the error you're seeing.

Since you're working with Elsa 3.0.0.0, you should also check the official documentation for any changes or updates in the way inputs are handled in this version.

If the problem persists, you might want to provide more details or check with the Elsa community or support channels for more specific guidance.

Remember to always test your workflows thoroughly before deploying them to production, especially when working with deadlines.

I hope this helps you resolve the issue! If you need further assistance, please provide additional details about your workflow and how you're executing it.
